### ISSUE TITLE
Rargb missing results

### ISSUE THREAD

**Issue by 89z** [pryme-svg/torrent-api#2]
Using this link:

https://torrent-api1.herokuapp.com/getTorrents?site=Rarbg&query=mollys+2017

I get several results, but no x265. While this page does list x265:

https://rarbg.to/torrents.php?search=mollys+2017

**Comment by pryme-svg** [comment]
I have noticed this problem as well, but the api simply does not return a complete list of results. It uses `https://torrentapi.org/pubapi_v2.php?mode=search&search_string={query}&token={token}&app_id=torrent-api&format=json_extended&sort=seeders`.

Obtain token here: https://torrentapi.org/pubapi_v2.php?get_token=get_token&app_id=ef

This would result in https://torrentapi.org/pubapi_v2.php?mode=search&search_string=mollys%202017&token=[token]&app_id=torrent-api&format=json_extended&sort=seeders after you replace the token, and it doesn't show x265.

![image](https://user-images.githubusercontent.com/65417452/120264744-47f14380-c253-11eb-9218-cd001ec7e9c8.png)

It is possible to crawl and scrape Rarbg, but it has a captcha system which makes it entirely impractical.

Unfortunately, unless the api returns the results that the site does, there is no way to fix this issue.

If you are interested, the documentation for the api is at https://torrentapi.org/apidocs_v2.txt?app_id=whatever_you_want_lol

**Comment by 89z** [comment]
> It is possible to crawl and scrape Rarbg, but it has a captcha system which makes it entirely impractical.

Not true. If you capture the Cookie, via HAR or similar, you can do it:

~~~
PS C:\> curl.exe -v -H 'Cookie: skt=6VXGP9wbmf' rarbg.to/torrents
*   Trying 185.37.100.122:80...
* Connected to rarbg.to (185.37.100.122) port 80 (#0)
> GET /torrents HTTP/1.1
> Host: rarbg.to
> User-Agent: curl/7.70.0
> Accept: */*
> Cookie: skt=6VXGP9wbmf
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Server: nginx
< Date: Tue, 01 Jun 2021 04:09:31 GMT
~~~

**Comment by 89z** [comment]
I figured it out, you just have to add the right category, in this case 54 (Movies/x265/1080):

https://torrentapi.org/pubapi_v2.php?search_string=mollys+2017&token=51o6g0csth&app_id=1&mode=search&category=44;54

**Comment by pryme-svg** [comment]
I enabled all categories and also added a `new` argument which scrapes the page instead of using the api. The api seems really unreliable and returns "No Result" at times. It might just be because the heroku ip is blacklisted. If it doesn't work, just add a `&new=true` to the url and it should return the proper results. For example, https://torrent-api1.herokuapp.com/getTorrents?site=Rarbg&query=mollys%202017&new=true

**Comment by 89z** [comment]
Thanks - I also had the issues with `No result`, I think their JSON API is pretty crappy. Also they host their own images:

<https://dyncdn.me/mimages/325299/poster_opt.jpg>

but they dont expose the images in the JSON API. The best I found was the `info_page` value:

https://torrentapi.org/redirect_to_info.php?token=51o6g0csth&p=2_2_1_2_9_5_9__aaf3f8243b&app_id=1

which just redirects to their HTML:

https://rarbg.to/torrent/qez2nco

so either way, you end up having to parse HTML. Or maybe since they give IMDb and TheMovieDb, you can get the pictures that way. I dont know.

**Comment by 89z** [comment]
Also, as I said before, this is probably not a long term solution:

https://github.com/pryme-svg/torrent-api/blob/c6a2d5963c0392efc9b1d1c318a74e55534b661a/scraper.py#L18

as I think it expires after 3 days:

~~~
Set-Cookie: skt=M8C3Y1mjxb; expires=Fri, 04-Jun-2021 18:47:17 GMT
~~~

**Comment by pryme-svg** [comment]
The API is just sort of hit or miss and the cookie for HTML parsing expires. I don't really know how to proceed with this unless you have any suggestions.

**Comment by pryme-svg** [comment]
It looks like replit's ips are not blacklisted, https://torrent-api.prymesvg.repl.co/getTorrents?site=Rarbg&query=mollys%202017 works pretty consistently.

**Comment by pryme-svg** [comment]
I pushed some changes, should work now

**Comment by 89z** [comment]
Hm, I dont know. I think this:

https://github.com/pryme-svg/torrent-api/blob/f0c8aae13a8e5250632c993500f8e0271419f3e5/scraper.py#L190

breaks the Cookie, as soon as as accessed your website with my Cookie, it broke my Cookie and I had to make a new one. Maybe too many requests.

**Comment by 89z** [comment]
I think that fixed it, nice

**Comment by pryme-svg** [comment]
It seems to be resolved. Can we close this now?

**Comment by 89z** [comment]
well, I dont know if youre interested to add this, but another thing I was looking at was MediaInfo:

~~~
PS C:\> curl -H 'X-Requested-With: XMLHttpRequest' -H 'Cookie: skt=Y4PQBeklfm' 'https://rarbg.to/ajax.php?mode=mediainfo&id=2212959_qez2nco'
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head></head><body>Video<br />
ID                                       : 1<br />
Format                                   : HEVC<br />
Format/Info                              : High Efficiency Video Coding<br />
Format profile                           : Main 10@L4@Main<br />
Codec ID                                 : hev1<br />
Codec ID/Info                            : High Efficiency Video Coding<br />
Duration                                 : 2 h 20 min<br />
Bit rate                                 : 2 000 kb/s<br />
Width                                    : 1 920 pixels<br />
Height                                   : 800 pixels<br />
~~~

**Comment by pryme-svg** [comment]
That doesn’t really fit my use case, so I’m not going to add it, but you can always initiate a pull request if you wish

**Comment by 89z** [comment]
Hm, seems like my Cookie expired after 12 hours. I will try it again today to confirm. If so, thats not good :(

**Comment by 89z** [comment]
Also I noticed that regarding bit rate, for another way to get it, you can take the `size`, then browse to the `info_page`, and divide by `IMDB Runtime`. This way you can make two requests instead of three.

**Comment by 89z** [comment]
If youre interested, I found a way to automate the login process, so you dont have to manually get the `skt` cookie. However my code is in Go, so you would need to translate it:

https://github.com/89z/torrent/blob/master/torrent.go

**Comment by pryme-svg** [comment]
repo no longer exists?

**Comment by 89z** [comment]
Its here now https://github.com/89z/mech/tree/master/rarbg
